.pre-commit-config.yaml

```diff
@@ -0,0 +1,44 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: trailing-whitespace
+        files: '\.py$'
+      - id: end-of-file-fixer
+        files: '\.py$'
+      - id: check-yaml
+      - id: check-added-large-files
+        files: '\.py$'
+
+  - repo: https://github.com/psf/black
+    rev: 24.3.0
+    hooks:
+      - id: black
+        language_version: python3.9
+        files: '\.py$'
+        args:
+          - --line-length=120
+          - --exclude=staging_service/auth2Client.py
+          - --exclude=import_specifications/clients/baseclient\.py
+
+  - repo: https://github.com/pycqa/flake8
+    rev: 7.0.0
+    hooks:
+      - id: flake8
+        args:
+          - --ignore=E203,W503
+          - --max-line-length=120
+          - --config=.flake8
+        files: '\.py$'
+        additional_dependencies: [ flake8 ]
+
+
+
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.4.1
+    hooks:
+      - id: ruff
+        args:
+          - '--exclude=tests/test_app.py'
```

Comment:
why exclude these tests?

Reply:
It was either exclude these tests or change every single single letter variable to be 3 letters as per the linting error that file gave

Reply:
removed the exclusion